libs/domain/cart/entry/src/entry.component.ts

```diff
@@ -1,170 +1,206 @@
+import { CartEntry } from '../../src/models';
 import {
-  BehaviorSubject,
-  combineLatest,
-  map,
-  merge,
-  Subject,
-  take,
-  tap,
-} from 'rxjs';
-import { CartEntryOptions, REMOVE_EVENT } from './entry.model';
-import { baseStyles, styles } from './styles';
-
-export class CartEntryComponent extends ComponentMixin<CartEntryOptions>() {
-  static styles = [baseStyles, styles];
+  CartEntryAttributes,
+  CartEntryChangeEventDetail,
+  CartEntryOptions,
+  RemoveByQuantity,
+} from './entry.model';
+import { cartEntryStyles } from './styles';
+
+/**
+ * Supports updating the quantity as well as removing the entry entirely.
+ */
+@defaultOptions({
+  removeByQuantity: RemoveByQuantity.ShowBin,
+  enableId: true,
+  enablePreview: true,
+} as CartEntryOptions)
+export class CartEntryComponent
+  extends CartComponentMixin(ContentMixin<CartEntryOptions>(LitElement))
+  implements CartEntryAttributes
+{
+  static styles = [cartEntryStyles];
+
+  @property() key?: string;
+  @property({ type: Boolean }) readonly?: boolean;
+  @property() available?: number;
+
+  @state() protected entry?: CartEntry;
 
   protected context = new ContextController(this);
-  protected options$ = new ContentController(this).getOptions();
-
-  protected triggerConfirmationRequired$ = new BehaviorSubject(false);
-  protected confirmation$ = combineLatest([
-    this.triggerConfirmationRequired$,
-    this.options$,
-  ]).pipe(
-    map(([confirmationRequired, entry]) => ({ confirmationRequired, entry }))
-  );
-
-  protected triggerShowOptions$ = new Subject<boolean>();
-  protected showOptions$ = merge(
-    this.triggerShowOptions$,
-    this.options$.pipe(
-      take(1),
-      map(({ defaultExpandedOptions }) => !!defaultExpandedOptions)
-    )
-  );
-
-  @subscribe()
-  protected sku$ = this.options$.pipe(
-    tap(({ sku }) => {
-      this.context.provide(ProductContext.SKU, sku);
-    })
-  );
-
-  protected toggleOptions(e: CustomEvent): void {
-    this.triggerShowOptions$.next(e.detail.state);
-  }
+  protected pricingService = resolve(PricingService);
 
-  protected dispatchRemoveEvent(): void {
-    this.dispatchEvent(
-      new CustomEvent(REMOVE_EVENT, { bubbles: true, composed: true })
-    );
-  }
-
-  protected onQuantityChange(
-    e: CustomEvent<QuantityEventDetail>,
-    isInstant: boolean
-  ): void {
-    if (e.detail.quantity === 0) {
-      e.stopPropagation();
+  @state() formattedPrice?: string;
 
-      if (isInstant) {
-        this.dispatchRemoveEvent();
-        return;
-      }
+  protected notificationService = resolve(NotificationService);
 
-      this.triggerConfirmationRequired$.next(true);
+  protected willUpdate(
+    changedProperties: PropertyValueMap<unknown> | Map<PropertyKey, unknown>
+  ): void {
+    this.entry = this.entries?.find((entry) => entry.groupKey === this.key);
+    if (this.entry) {
+      this.context.provide(ProductContext.SKU, this.entry.sku);
+
+      this.pricingService
```

Comment:
we should move it to separate subscription, but let's hope signals will make it better.
Still I'd add a TODO here, to cover this case later

Reply:
Mixins for the world!!

Reply:
we'll leave it for mixins, I'll create a ticket.